src/archive.cpp

```diff
@@ -184,6 +184,22 @@ namespace zim
     }
   }
 
+  Entry Archive::getRandomEntry() const {
+    auto frontEntryCount = m_impl->getFrontEntryCount().v;
+    int watchdog = 42;
+    while(--watchdog) {
+      auto idx = randomNumber(frontEntryCount);
```

Comment:
`frontEntryCount` → `frontEntryCount-1` (and handle the case of an empty ZIM archive)

Reply:
Good catch